Add forwarding example

csmeyns opened the pull request:
This change adds an example for the 'How to add forwarding to your chat' tutorial, which is still to be published. A link to the tutorial should still be added in the `README.md` here once it's available ([PR here](https://github.com/talkjs/tutorials/pull/22)). Originally a guide for this was requested in this [Slack conversation](https://talkjs.slack.com/archives/C1ZJ9GGLA/p1725606938196999).

For info and reviewing: 
- I opted to give the full instructions (also for the Theme styling steps) here, rather than link to the tutorial, so that the user doesn't have to switch between different sets of instructions. That's why the instruction steps take up a bit more space than in some other examples.
- As discussed [on Slack](https://talkjs.slack.com/archives/CM78GKPED/p1729833979830519), this guide has the limitation that it only works for forwarding text-based messages. This limitation has been listed explicitly in the Readme as well.

keerlu commented on howtos/how-to-add-forwarding/README.md:
````diff
@@ -0,0 +1,75 @@
+        <!-- Check for forwarded messages -->
+        <div class="{{ custom.forwardedFrom | then: 'forwarded' }}"></div>
+        ```
+    3. Then, to style the message body of a forwarded message, go to `MessageBody` and add the following inside the `<template>` section:
````
I put it at the end, is that correct?

csmeyns replied:
In this case the location does make a difference, as it'll determine where the label appears. So I've clarified that this should be all the way at the beginning of the template. Thanks so much for bringing this up 🙂

keerlu reviewed (commented):
Just trying this out... got a couple of minor comments/questions on the README, but the other thing is that the modal isn't populated when I run it, presumably because there aren't any other conversations?

If that's the issue, you should be able to add something like [the setup code here](https://github.com/talkjs/talkjs-examples/blob/master/howtos/how-to-delete-a-conversation/server.js#L34) to your backend code to add a couple of example conversations ready to go.

I'll review the code properly once I've got the example working, but from a quick look it looks good and the UI design is really nice

<img width="849" alt="image" src="https://github.com/user-attachments/assets/707dc131-e26e-4345-ab25-6f02cfd3430d">

csmeyns commented on howtos/how-to-add-forwarding/README.md:
````diff
@@ -0,0 +1,75 @@
+3. Add styling to distinguish forwarded messages in your theme. The following steps add a subtle but noticeable right-pointed arrow and the label *Forwarded* at the top of the message body of messages that have been forwarded.
+    1. On the **Themes** page of your dashboard, select **Edit** for the theme that you’d like to use.
+    2. To identify user messages that have been forwarded, in `UserMessage`, add the following to the `<template>` part of the code:
+        ```html
````
Thanks for flagging this! I tried out different locations and this didn't seem to make any difference, so now as a pointer I've suggested that people add it at the top.

keerlu reviewed (commented):
Couple of tiny things here - otherwise mentioned stuff on Slack

csmeyns commented on howtos/how-to-add-forwarding/server.js:
```diff
@@ -0,0 +1,236 @@
+    	}
+	];
+
+	console.log(requestBody);
```
Thanks for spotting! Yes that one can go 👌

csmeyns commented:
Updated as follows:
- Make sure that the current conversation properly switches when the user selects a different conversation.
- Updated the instructions on how to edit your Theme to make the forward-specific styling show up.
- Added a note to the README to properly install Tailwind if you'd like to use it in production.
- Updated that ARIA labelling to: (1) not use `aria-hidden` for elements that are focusable. (2) Added an extra aria label for clarity. Note: I didn't get the specific warning about ARIA you shared in my own browser console, so may have to update some alerting settings here. But if all is well this change should address the issue.

Note: These changes have at the point of writing this only been made in this example. I'll now also add them to the tutorial instructions 🙂 

@keerlu if you wouldn't mind checking if this now works as expected? Thanks so much!

keerlu commented on howtos/how-to-add-forwarding/README.md:
````diff
@@ -0,0 +1,92 @@
+    1. On the **Themes** page of your dashboard, select **Edit** for the theme that you’d like to use.
+    2. To identify user messages that have been forwarded, in `UserMessage`, add the following inside the `<template>` part of the code, for example at the top:
+        ```html
+        <!-- Check for forwarded messages -->
````
It would be good to add a check that the `forwardedFrom` custom property exists. I got a. "Cannot read property 'forwardedFrom' in expression 'custom.forwardedFrom'" console error when I tried swapping to a new conversation in the frontend to see what happened (functionality still worked fine though)

csmeyns replied:
Thanks for the suggestion! Will do that 👌

keerlu reviewed (approved):
Nice, just tested and conversation switching is working well now! One suggestion for a check in the theme editor

keerlu commented on howtos/how-to-add-forwarding/server.js:
```diff
@@ -0,0 +1,236 @@
+app.use(cors());
+
+// Replace with your own app ID and secret key
+const appId = '<APP_ID>'
+const secretKey = '<SECRET_KEY>'
```
```suggestion
const appId = '<APP_ID>';
const secretKey = '<SECRET_KEY>';
```

csmeyns commented:
Thanks so much! For some reason I didn't manage get the error, so wasn't able to test when it would or wouldn't arise. In any case will update it like that 🙏
